### PR TITLE
add trmv LAPACK wrapper

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

**Comment by davydden** [comment]
./run-tests

**Comment by bangerth** [comment]
Would it be easy to add a test? The function isn't used anywhere right now, so it'd be real easy to break the interface :-)

**Comment by davydden** [comment]
> Would it be easy to add a test? The function isn't used anywhere right now, so it'd be real easy to break the interface :-)

it is, but i don't see a point. When it will be used and we have wrappers from `LAPACKFullMatrix`, then the test is in order.

**Comment by davydden** [comment]
@bangerth nevermind, I will add a few things to this PR in addition to the current changes....

File: include/deal.II/lac/lapack_full_matrix.h [hunk 1]
```diff
@@ -444,6 +444,12 @@ class LAPACKFullMatrix : public TransposeTable<number>
    */
   number reciprocal_condition_number(const number l1_norm) const;
 
+  /**
+   * Same as above but for triangular matrices. The matrix has to have
+   * the LAPACKSupport::Property set, see set_property().
+   */
+  number reciprocal_condition_number() const;
```
Comment (bangerth) [inline include/deal.II/lac/lapack_full_matrix.h]: You say "same as above", but what norm is this then computed for?

Does the property have to be set to something specific?

**Comment by davydden** [comment]
@bangerth please give it another look, I added much more things. Will fix docu later...

File: include/deal.II/lac/lapack_full_matrix.h [hunk 2]
```diff
@@ -478,14 +484,34 @@ class LAPACKFullMatrix : public TransposeTable<number>
...
 
+  /**
+   * Solve the linear system with right hand side. The matrix should
+   * be either triangular or LU factorization should be previously computed.
+   *
+   * The flag transposed indicates whether the solution of the transposed
+   * system is to be performed.
+   */
+  void solve(Vector<number> &v,
+             const bool transposed) const;
```
Comment (bangerth) [inline include/deal.II/lac/lapack_full_matrix.h]: maybe give it a default of `false`? That's the usual case.

Also, can you state that the solution is put back into the vector w?

File: include/deal.II/lac/lapack_full_matrix.h [hunk 3]
```diff
@@ -478,14 +484,34 @@ class LAPACKFullMatrix : public TransposeTable<number>
...
+
+  /**
+   * Same as above but for multiple right hand sides (as many as there
+   * are columns in the matrix @p B).
+   */
+  void solve(LAPACKFullMatrix<number> &B,
+             const bool transposed) const;
```
Comment (bangerth) [inline include/deal.II/lac/lapack_full_matrix.h]: same here

**Comment by davydden** [comment]
@bangerth I addressed your comments.

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

**Pull request merged**